### ISSUE TITLE
sanity_log_dead not implemented in sr3

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#626]
As part of sr audit  in v2 there was a check to see if the log was no longer being written to... indicating that the process is hung.
the sanity_log_dead was a duration setting, so ... if a log had not been modified in, say, 15 minutes, sr sanity run would emit an error message like 
```

 self.logger.warning("logfile older than %d; restarting %s" % (self.sanity_log_dead, self.instance_str))
        self.restart_instance()

```

and restart the instance in question.

This check and fix is not implemented in sr.

**Comment by petersilva** [comment]
worth looking at metrics from #614 also... as the age of the metrics file is a pertinent judge of whether an instance is stuck.

**Comment by petersilva** [comment]
perhaps look at doing something reasonable with:

```

                        if self.states[c][cfg]['instance_metrics'][i]['status']['mtime'] < expiry:
                            #print( f"metrics for {c}/{cfg}/ instance {i} too old ignoring." )
                            continue

```
in sarracenia/sr.py ... perhaps sanity could remove obsolete metrics... or help diagnose... "last diagnostics reported at..." or something...

**Comment by petersilva** [comment]
also noticed this... def  ageoffile(): stub that was not implemented... checking for the age of log files.
this commit: f0bd4fed3bc3aa5d1d05cf944c581c680fa97517

**Comment by petersilva** [comment]
released with v3.00.37
